napari/_vispy/vispy_base_layer.py

```diff
@@ -66,14 +71,66 @@ def __init__(self, layer, node):
     @property
     def _master_transform(self):
         """vispy.visuals.transforms.MatrixTransform:
+
         Central node's firstmost transform.
         """
         # whenever a new parent is set, the transform is reset
         # to a NullTransform so we reset it here
-        if not isinstance(self.node.transform, MatrixTransform):
-            self.node.transform = MatrixTransform()
+        if isinstance(self.node.transform, NullTransform):
+            self.node.transform = ChainTransform(
+                [STTransform(), MatrixTransform()]
+            )
+
+        return self.node.transform.transforms[1]
+
+    @property
+    def _grid_transform(self):
+        """vispy.visuals.transforms.MatrixTransform:
+
+        Transform used if viewer is in grid mode
+        """
+        # whenever a new parent is set, the transform is reset
+        # to a NullTransform so we reset it here
+        if isinstance(self.node.transform, NullTransform):
+            self.node.transform = ChainTransform(
+                [STTransform(), MatrixTransform()]
+            )
+
+        return self.node.transform.transforms[0]
```

Comment:
@sofroniewn to me it's still pretty concerning that the grid transform is not in the layer but it *is* in the VisPy layer...